src/ui_dialogs.ts

```diff
@@ -115,17 +164,270 @@ export function openEditLinkDialog(column:LinkColumn, $header:d3.Selection<Colum
   });
 }
 
+
+// Renderer type change
+export function renderertypedialog(column: Column, $header: d3.Selection<Column>) {
+  var renderertype = (<any>column.desc).type;
+  var valuestring: any = (<any>column.desc).renderertype;
+  var renderername: any = [];
+  valuestring.forEach(function (d) {
+    if (d === 'heatmapcustom') {
+      renderername.push('Heatmap');
+    }
+    if (d === 'sparkline') {
+      renderername.push('Sparkline');
+    }
+    if (d === 'threshold') {
+      renderername.push('Threshold');
+    }
+    if (d === 'upset') {
+      renderername.push('Upset');
+    }
+    if (d === 'circle') {
+      renderername.push('Circle');
+    }
+    if (d === 'number') {
+      renderername.push('Bar');
+    }
+    if (d === 'boxplot') {
+      renderername.push('Box Plot');
+    }
+    if (d === 'verticalbar') {
+      renderername.push('Vertical Bar');
+    }
+  });
+
+  var popup = makesortPopup($header, 'Change Visualization', valuestring.map(function (d, i) {
+    return `<br><input type="radio" name="renderertype" value=${d}  ${(renderertype === d) ? 'checked' : ''}> ${renderername[i]}`;
+
+  }).join('\n'));
+
+  function thiselement() {
+    return this === (<any>d3.event).target;
+
+  }
+
+  var that;
+
+  var renderercontent = d3.selectAll('input[name="renderertype"]');
+
+  renderercontent.on('change', function () {
+    that = this;
+    renderertype = that.value;
+    column.setRendererType(that.value)
+
+    // column.setMetaData({
+    //   label: column.getMetaData().label,
+    //   description: column.getMetaData().description,
+    //   color: column.getMetaData().color
+    // });
+
+  });
+
+  d3.select('body').on('click', function () {
+    var outside = renderercontent.filter(thiselement).empty();
+    if (outside) {
+      popup.remove();
+    }
+  });
+}
+
+// Sort Heatmap Dialog.
+export function sortDialogHeatmap(column: HeatmapColumn, $header: d3.Selection<HeatmapColumn>) {
+  var rank = (<any>column.desc).sort;
+  const valuestring: any = ['min', 'max', 'mean', 'median', 'q1', 'q3'];
+  const sortlabel: any = ['Min', 'Max', 'Mean', 'Median', 'Q1', 'Q3'];
+
+  var popup = makesortPopup($header, 'Sort By', valuestring.map(function (d, i) {
+    return `<br><input type="radio" name="heatmaprank" value=${d}  ${(rank === d) ? 'checked' : ''} > ${sortlabel[i]}`;
+
+  }).join('\n'));
+
+  function thiselement() {
+
+    return this === (<any>d3.event).target;
+  }
+
+  var that;
+
+  var sortcontent = d3.selectAll('input[name=heatmaprank]');
+  sortcontent.on('change', function () {
+    that = this;
+    rank = that.value;
+    (<any>column.desc).sort = rank;
+    column.toggleMySorting();
```

Comment:
toggleMySorting = sorts by myself and reverses the ordering: asc -> desc and vice versa. However, the sorting order shouldn't change